src/runtime_src/core/tools/xbmgmt2/SubCmdProgram.cpp

```diff
@@ -336,12 +345,20 @@ updateShellAndSC(unsigned int  boardIdx, DSAInfo& candidate, bool& reboot)
   if (!same_bmc) {
     std::cout << "Updating SC firmware on device[" << flasher.sGetDBDF() <<
       "]" << std::endl;
-    update_SC(boardIdx, candidate.file);
+    try {
+      update_SC(boardIdx, candidate.file);
+    } catch (const xrt_core::error& e) {
+      std::cout << "NOTE: Skipping SC flash. " << e.what() << std:: endl;
+    }
   }
 
   if (!same_dsa) {
     std::cout << boost::format("[%s] : Updating base\n") % flasher.sGetDBDF();
-    update_shell(boardIdx, candidate.file, candidate.file);
+    try {
+      update_shell(boardIdx, candidate.file, candidate.file);
+    } catch (const xrt_core::error& e) {
+      std::cout << "NOTE: " << e.what() << std:: endl;
```

Comment:
It is always good to look at the error being caught if it is being ignored.  If it isn't what is expected then it should be re-thrown.

Reply:
That is what is being done here. We want to look at the error produced but we don't want to stop the auto-update. If one device fails, we want to look at it, but let the other devices continue to update